backends-clickhouse/src/main/delta-22/org/apache/spark/sql/delta/DeltaLog.scala

```diff
@@ -0,0 +1,918 @@
+      // conflict with `DeltaLog.options`.
+      snapshotToUse.metadata.format.options ++ options
+    )(
+      spark
+    ) with InsertableRelation {
+      def insert(data: DataFrame, overwrite: Boolean): Unit = {
+        val mode = if (overwrite) SaveMode.Overwrite else SaveMode.Append
+        WriteIntoDelta(
+          deltaLog = DeltaLog.this,
+          mode = mode,
+          new DeltaOptions(Map.empty[String, String], spark.sessionState.conf),
+          partitionColumns = Seq.empty,
+          configuration = Map.empty,
+          data = data
+        ).run(spark)
+      }
+    }
+  }
+
+  /**
+   * Verify the required Spark conf for delta Throw
+   * `DeltaErrors.configureSparkSessionWithExtensionAndCatalog` exception if
+   * `spark.sql.catalog.spark_catalog` config is missing. We do not check for `spark.sql.extensions`
+   * because DeltaSparkSessionExtension can alternatively be activated using the `.withExtension()`
+   * API. This check can be disabled by setting DELTA_CHECK_REQUIRED_SPARK_CONF to false.
+   */
+  protected def checkRequiredConfigurations(): Unit = {
+    if (spark.sessionState.conf.getConf(DeltaSQLConf.DELTA_REQUIRED_SPARK_CONFS_CHECK)) {
+      if (spark.conf.getOption(SQLConf.V2_SESSION_CATALOG_IMPLEMENTATION.key).isEmpty) {
+        throw DeltaErrors.configureSparkSessionWithExtensionAndCatalog(None)
+      }
+    }
+  }
+
+  /**
+   * Returns a proper path canonicalization function for the current Delta log.
+   *
+   * If `runsOnExecutors` is true, the returned method will use a broadcast Hadoop Configuration so
+   * that the method is suitable for execution on executors. Otherwise, the returned method will use
+   * a local Hadoop Configuration and the method can only be executed on the driver.
+   */
+  private[delta] def getCanonicalPathFunction(runsOnExecutors: Boolean): String => String = {
+    val hadoopConf = newDeltaHadoopConf()
+    // Wrap `hadoopConf` with a method to delay the evaluation to run on executors.
+    val getHadoopConf = if (runsOnExecutors) {
+      val broadcastHadoopConf =
+        spark.sparkContext.broadcast(new SerializableConfiguration(hadoopConf))
+      () => broadcastHadoopConf.value.value
+    } else { () => hadoopConf }
+
+    new DeltaLog.CanonicalPathFunction(getHadoopConf)
+  }
+
+  /**
+   * Returns a proper path canonicalization UDF for the current Delta log.
+   *
+   * If `runsOnExecutors` is true, the returned UDF will use a broadcast Hadoop Configuration.
+   * Otherwise, the returned UDF will use a local Hadoop Configuration and the UDF can only be
+   * executed on the driver.
+   */
+  private[delta] def getCanonicalPathUdf(runsOnExecutors: Boolean = true): UserDefinedFunction = {
+    DeltaUDF.stringFromString(getCanonicalPathFunction(runsOnExecutors))
+  }
+
+  override def fileFormat(metadata: Metadata = metadata): FileFormat =
+    ClickHouseTableV2.deltaLog2Table(this).getFileFormat(metadata)
```

Comment:
it seems it can not use this way to get the `ClickHouseTableV2`, because if there is no writing data operation in this spark session, right?